candid/evm_rpc.did

```diff
@@ -273,7 +263,5 @@ service : (InitArgs) -> {
   request : (RpcService, json : text, maxResponseBytes : nat64) -> (RequestResult);
   requestCost : (RpcService, json : text, maxResponseBytes : nat64) -> (RequestCostResult) query;
   setOpenRpcAccess : (active : bool) -> ();
-  unregisterProvider : (ProviderId) -> (bool);
-  updateProvider : (UpdateProviderArgs) -> ();
```

Comment:
This is a breaking change. It's worth noting that `do_update_provider` and `do_unregister_provider` are public functions, so deleting these would be another breaking change to anyone using the EVM RPC canister as a crate. This seems unlikely, although it might be useful to keep this functionality around in case anyone forks the canister and wants to reintroduce ways of updating / removing providers. 

I'll remove these methods for the time being. I suppose another option could be to restrict the permission to the canister controller if you think this would be preferable. Any change to mutability seems like it would necessarily be a breaking change for RPC provider management.

Reply:
Understanding question about the implications of this PR: does-it mean that if some principal already registered a custom provider, then that principal can no longer change it? If that's the case, then maybe it's something worth discussing this PR a bit more deeply (zoom), since basically the ckETH minter only needs the providers which are hard-coded as `EthMainnetService` and `EthSepoliaService` to be immutable. In general, I think it makes sense that the "officially" supported providers by the EVM-RPC canister (`EthMainnetService`, `EthSepoliaService` and `L2MainnetService`) are immutable, but for custom registered ones I think it's totally ok if there are not.

Reply:
> but for custom registered ones I think it's totally ok if there are not.

Does this refer to providers which are defined by users rather than DFINITY? The only providers which are registered in the canister are "official" in the sense that they are directly used for `EthMainnetService`, `EthSepoliaService`, and `L2MainnetService`. For user-defined providers, all information is passed at call-time. 

Is your suggestion to keep mutability for providers in general (with the assumption that we will enable user-specified providers in the future) but enforce immutability for DFINITY-owned providers? We could continue talking about this over Zoom if that's preferable.

Reply:
I sent an invite since I think it makes sense to discuss a bit what exactly we need to do here. My worry is that we are 
 maybe trying to do too much in a rush, but maybe that's just due to my incorrect understanding of how the permissions and the registration of providers work in the EVM RPC canister. I think it makes sense to answer your questions and mine when we (virtually) meet, since doing via comments is generally not that efficient.